speaker/layer2_controller.go

```diff
@@ -133,9 +134,24 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 	return "notOwner"
 }
 
+func getInterfaces(localNode string, l2Advertisements []*config.L2Advertisement) sets.String {
+	ifs := sets.NewString()
+	for _, l2 := range l2Advertisements {
+		if matchNode, ok := l2.Nodes[localNode]; !ok || !matchNode {
+			continue
+		}
+		if len(l2.Interfaces) == 0 {
```

Comment:
Can you add a comment here explaining the semantic? I am also tempted to wrap the set under a 
struct {
   allInterfaces bool
   interfaces Set
}

WDYT? I am not sure I like propagating the len==0 -> all interfaces down there. A flag might be more explicit and understandable.
Ideally, that flag could be even a property of the "internal" l2advertisement, so the only place where we consider  len==0 -> all interfaces is the CRD.

Reply:
ok